crates/ra_ide/src/hover.rs

```diff
@@ -133,7 +106,7 @@ fn determine_mod_path(db: &RootDatabase, def: &Definition) -> Option<String> {
             .flatten()
             .join("::")
     });
-    mod_path
+    mod_path // FIXME: replace dashes with underscores in crate display name
```

Comment:
@SomeoneToIgnore this appears to be syntax error (though not a critical one) since the path is highlighted as rust source code we get the dash highlighted as a subtractions operator:
![image](https://user-images.githubusercontent.com/36276403/76714487-e0bfa000-672f-11ea-9364-43a381be3dfb.png)

Should we just post-process the mod_path by replacing dashes with underscores?

Reply:
* Or should we replace dashes with underscores only for crate's `display_name` here?
* Or should we preprocess the `display_name` and replace dashes with underscores when it is assigned to crate data?

Reply:
There was a similar issue solved by the: https://github.com/rust-analyzer/rust-analyzer/blob/c48dcf74118b6e0df747f036a9b66701037f3fc7/crates/ra_db/src/input.rs#L101
method and related changes.

I think it would be good to find the source of those dashes and reuse the code to fix them.

Reply:
https://github.com/rust-analyzer/rust-analyzer/blob/c48dcf74118b6e0df747f036a9b66701037f3fc7/crates/ra_db/src/input.rs#L113

This should be the `Option<CrateName>` instead.

Reply:
Hmm... I wonder maybe there was some rationale not to use it for `display_name`?

Reply:
I simply did not think about this when was using `Option<String>` instead.